packages/mobile/App/ui/components/MultiSelectModal/MultiSelectModalField.tsx

```diff
@@ -41,35 +45,42 @@ export const MultiSelectModalField = ({
   const navigation = useNavigation();
   const [label, setLabel] = useState(null);
 
-  const appendLabel = (items: OptionType[]) => {
-    return items.map(x => ` ${x.label}`).toString();
-  };
-  const onPress = (selectedItems: OptionType[]): void => {
+  const handleSaveCallback = (selectedItems: OptionType[]): void => {
     onChange(selectedItems);
-    setLabel(appendLabel(selectedItems));
+    setLabel(extractLabel(selectedItems));
   };
 
   const openModal = (): void =>
     navigation.navigate(modalRoute, {
-      callback: onPress,
+      callback: handleSaveCallback,
       suggester,
       modalTitle,
       value,
       searchPlaceholder,
     });
-  const loadDefaultValues = async (values: string[]) => {
-    const _values: OptionType[] = [];
-    values.forEach(async x => {
+
+  // This function is not in use, however, it's meant to be used when
+  // initial values are set. Pay extra care to the value format it expects.
```

Comment:
I'm a bit confused about this comment as this function is being used below right?

Reply:
ok that sounds good to me